check/src/project.rs

```diff
@@ -161,6 +166,32 @@ fn all_paths(root_dir: &Path, source_file_patterns: Vec<String>) -> Result<Vec<P
     Ok(files)
 }
 
+pub fn extract_toolchain_channel(toolchain_file_path: &PathBuf) -> Result<String> {
+    let toolchain_file_contents = std::fs::read_to_string(toolchain_file_path).wrap_err(
+        "expected to find a rust-toolchain.toml file in project directory \
+         to specify your Rust toolchain for reproducible verification",
+    )?;
+    let toolchain_toml: Value =
+        toml::from_str(&toolchain_file_contents).wrap_err("failed to parse rust-toolchain.toml")?;
+
+    // Extract the channel from the toolchain section
+    let Some(toolchain) = toolchain_toml.get("toolchain") else {
+        bail!("toolchain section not found in rust-toolchain.toml");
+    };
+    let Some(channel) = toolchain.get("channel") else {
+        bail!("could not find channel in rust-toolchain.toml's toolchain section");
+    };
+    let Some(channel) = channel.as_str() else {
+        bail!("channel in rust-toolchain.toml's toolchain section is not a string");
+    };
+    // Next, parse the Rust version from the toolchain project, only allowing alphanumeric chars and dashes.
+    let channel = channel
+        .chars()
+        .filter(|c| c.is_alphanumeric() || *c == '-')
```

Comment:
It looks like `.` should also be a valid character according to the docs. `<channel>       = stable|beta|nightly|<major.minor.patch>`

Reply:
Thanks for confirming ! will add